### PR TITLE
Implement Blame Taint Scopes

### PR DESCRIPTION
**Pull request by boehmseb**
Blame Taints now can have 4 different scopes:
 - GLOBAL
 - REGION
 - COMMIT_IN_FUNCTION
 - COMMIT

Therefore, the old `GenerateBlameReport` experiment gets split into three variants: `region`, `commit_in_function`, and `commit` with the last one corresponding to the old report.

### CODE REVIEW

File: varats/varats/experiments/vara/blame_report_experiment.py [hunk 1]
```diff
@@ -130,8 +137,33 @@ def actions_for_project(
...
+
+    NAME = "GenerateBlameReportCommitInFunction"
+    BLAME_TAINT_SCOPE = BlameTaintScope.COMMIT_IN_FUNCTION
+
+
+class BlameReportExperimentCommit(BlameReportExperiment, shorthand="BREC"):
+    """Generates a blame report with commit scoped taints."""
+
+    NAME = "GenerateBlameReportCommit"
+    BLAME_TAINT_SCOPE = BlameTaintScope.COMMIT
```
Comment (vulder) [inline varats/varats/experiments/vara/blame_report_experiment.py]:
```suggestion
```

File: varats/varats/data/reports/blame_report.py [hunk 2]
```diff
@@ -335,6 +448,14 @@ def __init__(
         self.__base_head = base_report.head_commit
         self.__prev_head = prev_report.head_commit
         self.__calc_diff_br(base_report, prev_report)
+        assert \
```
Comment (vulder) [inline varats/varats/data/reports/blame_report.py]: Maybe use `raise AssertionError`

File: varats/varats/data/reports/blame_report.py [hunk 3]
```diff
@@ -21,85 +22,169 @@
...
+    def function_name(self) -> tp.Optional[str]:
+        return self.__function_name
+
+    @property
+    def commit(self) -> CommitRepoPair:
+        return self.__commit
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, BlameTaintData):
+            return NotImplemented
```
Comment (vulder) [inline varats/varats/data/reports/blame_report.py:73]: Fix me

File: varats/varats/data/reports/blame_report.py [hunk 4]
```diff
@@ -21,85 +22,169 @@
...
+
+        return not ((self.region_id != other.region_id) or
+                    (self.function_name != other.function_name) or
+                    (self.commit != other.commit))
+
+    def __lt__(self, other: object) -> bool:
+        if not isinstance(other, BlameTaintData):
+            return NotImplemented
```
Comment (vulder) [inline varats/varats/data/reports/blame_report.py:107]: Fix me

**Review by vulder — APPROVED** [review]

File: varats/varats/data/reports/blame_report.py (hunk 3 above)
Comment (boehmseb) [reply varats/varats/data/reports/blame_report.py:73]: This is actually a python feature: https://docs.python.org/3/library/constants.html#NotImplemented

File: varats/varats/data/reports/blame_report.py (hunk 4 above)
Comment (boehmseb) [reply varats/varats/data/reports/blame_report.py:107]: same as above

**Pull request merged**